src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -154,12 +155,19 @@ static DiagnosticInfo createObsoleteDiagnostic(Symbol symbol, BinderFlags locati
                     return null;
                 }
 
-                if (data.Kind == ObsoleteAttributeKind.Experimental)
+                if (data.Kind == ObsoleteAttributeKind.WindowsExperimental)
                 {
                     Debug.Assert(data.Message == null);
                     Debug.Assert(!data.IsError);
-                    // Provide an explicit format for fully-qualified type names.
-                    return new CSDiagnosticInfo(ErrorCode.WRN_Experimental, new FormattedSymbol(symbol, SymbolDisplayFormat.CSharpErrorMessageFormat));
+                    return new CSDiagnosticInfo(ErrorCode.WRN_Experimental, symbol);
+                }
+
+                if (data.Kind == ObsoleteAttributeKind.Experimental)
+                {
+                    Debug.Assert(data.Message is null);
+                    Debug.Assert(!data.IsError);
+
+                    return new CustomObsoleteDiagnosticInfo(MessageProvider.Instance, (int)ErrorCode.WRN_Experimental, data, symbol);
```

Comment:
>CustomObsoleteDiagnosticInfo [](http://example.com/codeflow?start=31&length=28)

Why use `CustomObsoleteDiagonsticInfo` rather than `CSDiagnosticInfo`?